src/hotspot/share/opto/superword.cpp

```diff
@@ -2024,6 +2024,13 @@ bool SuperWord::implemented(Node_List* p) {
         retValue = ReductionNode::implemented(opc, size, arith_type->basic_type());
       }
     } else {
+      // Vector unsigned right shift for signed subword types behaves differently
+      // from Java Spec. But when the shift amount is a constant not greater than
+      // the number of sign extended bits, the unsigned right shift can be
```

Comment:
I'm still not clear why `>>>` can be transferred to `>>` iff `shift_cnt <= the number of sign extended bits`.

Reply:
When shift_cnt <= the number of sign extended bits, vector signed right shift can work the same as scalar unsigned right shift for subword types in the case I mentioned above, when short value is negative. Vector right shift can fill the shifted bits with sign-extension as we expected. As for positive short values, n >>> s  works the same as n >> s in Java spec.